### PR TITLE
Added libmagic fallback for filetype detection

### LINKED ISSUE #40: if no file extension, infer language from shebang, if any

### PR DESCRIPTION
**Pull request by cmlsharp**
Fixes #40

### CODE REVIEW

File: style50/style50.py [hunk 1]
```diff
@@ -179,13 +182,17 @@ def _check(self, file):
...
-
-            with open(file) as f:
-                code = f.read()
         except KeyError:
-            raise Error("unknown file type \"{}\", skipping...".format(file))
-        else:
-            return check(code)
+            magic_type = magic.from_file(file)
+            for names, cls in self.magic_map.items():
+                if any(name in magic_type for name in names):
```
Comment (dmalan) [inline style50/style50.py:188]: Does `any` short-circuit these iterations? If not, is intersection cleaner? Just curious.

File: style50/style50.py [hunk 2]
```diff
@@ -304,6 +313,9 @@ class StyleCheck(object):
     # Warn if fewer than 10% of code is comments.
```
Comment (dmalan) [inline style50/style50.py]: Technically "less".

File: style50/style50.py [hunk 3]
```diff
@@ -17,6 +17,7 @@
 
 import icdiff
 import six
+import magic
```
Comment (dmalan) [inline style50/style50.py]: We've generally alphabetized.

**Review by dmalan — APPROVED** [review]

File: style50/style50.py (hunk 1 above)
Comment (cmlsharp) [reply style50/style50.py:188]: Yes, any short circuits.

File: style50/style50.py (hunk 2 above)
Comment (cmlsharp) [reply style50/style50.py]: My bad, will fix

**Pull request merged**